pkg/edit/state_api.go

```diff
@@ -43,6 +47,10 @@ func initStateAPI(app cli.App, nb eval.NsBuilder) {
 		if err != nil {
 			return err
 		}
+		state := codeArea.CopyState()
+		if dot < 0 || dot > len(state.Buffer.Content) {
+			return errDotOutOfBoundary
+		}
```

Comment:
This should be part of the `MutateState` block below, otherwise the state may have changed in between.